ml_tools/trackextractor.py

```diff
@@ -796,24 +834,34 @@ def get_regions_of_interest(self, filtered, prev_filtered=None):
                 stats[i, 4], 0, i, self.frame_on
             )
 
-
-            if self.include_cropped_regions:
+            old_region = region.copy()
+            if self.CROPPED_REGIONS_STRATEGY == "all":
                 # in this case we just clip the region to the edges of the screen
-                old_region = str(region)
                 region.left = max(region.left, 0)
                 region.top = max(region.top, 0)
-                region.right = min(region.right, frame_width-1)
-                region.bottom = min(region.bottom, frame_height-1)
-                region.was_cropped = old_region != str(region)
-            else:
-                # if region went outside if bounds ignore it.
+                region.right = min(region.right, frame_width - 1)
+                region.bottom = min(region.bottom, frame_height - 1)
+            elif self.CROPPED_REGIONS_STRATEGY == "cautious":
+                # keep cropped regions if they have a reasonable size
+                region.left = max(region.left, 0)
+                region.top = max(region.top, 0)
+                region.right = min(region.right, frame_width - 1)
+                region.bottom = min(region.bottom, frame_height - 1)
+
+                crop_width_fraction = (old_region.width - region.width) / old_region.width
+                crop_height_fraction = (old_region.height - region.height) / old_region.height
+
+                if crop_width_fraction > 0.25 or crop_height_fraction > 0.25: continue
+
+            elif self.CROPPED_REGIONS_STRATEGY == "none":
+                # all regions that touch the side of the screen are removed
                 if (region.left < 0) or (region.right > frame_width) or (region.top < 0) or (region.top < 0) or (region.bottom > frame_height):
                     continue
-
-            # if region is too large ignore it.  this is mostly because very large objects do not track well,
-            # and take up a lot of space.
-            if (region.width > self.MAX_REGION_SIZE) or (region.height > self.MAX_REGION_SIZE):
-                continue
+            else:
+                raise Exception(
```

Comment:
ValueError would be an appropriate and more specific exception type to use here

https://docs.python.org/3/library/exceptions.html#ValueError